src/clinvar_streams/storage/database_sqlite/sink.clj

```diff
@@ -637,10 +637,64 @@
   (let [release-date (:release_date release-sentinel)
         get-simple (fn [table-name]
                      (let [s (format "select * from %s where dirty = 1 and release_date = ?" table-name)
-                           u (format "update %s set dirty = 0 where release_date = ?" table-name)
+                           ;u (format "update %s set dirty = 0 where release_date = ?" table-name)
                            rs (query @db-client/db [s release-date])]
                        ;(execute! @db-client/db [u release-date])
-                       rs))]
+                       rs))
+        get-variation (fn []
```

Comment:
@theferrit32 is this normally done like this? this looks like a nested function. Should it be a top level function? Just trying to learn a little here. I really don't understand much of the code still ;)

Reply:
@larrybabb I see it both ways in some of the Clojure code that I read. Nested functions can be nicer for having smaller scope, but can get you some very long function signatures (as with this one, esp with the embedded SQL). Also they can be closures using local vars (as this one is).

As for understanding the code, the key is understanding Kyle's system of flagging updated records as 'dirty' during an update. On this front I'll admit I'm a little behind as well.

Reply:
@larrybabb I originally just had `get-simple` and as a nested function because it is only really useful in this context. `get-variation` then was added because variations are a special case, and that turned into a bit more code than I was hoping so then number of lines in the function is a bit longer than ideal. Lines 674-695 might be generalizable to nested records like clinical assertions so might be pulled out later because those will need the release dates cleaned up too.